pallets/migration/src/lib.rs

```diff
@@ -76,11 +76,22 @@ pub mod pallet {
 
 		/// WeightInfo
 		type WeightInfo: WeightInfo;
+
+		type Filter: Contains<<Self as frame_system::Config>::Call>;
 	}
 
 	#[pallet::hooks]
 	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
 
+	#[pallet::type_value]
+	pub fn OnCompletedEmpty() -> bool {
+		false
+	}
+
+	#[pallet::storage]
+	#[pallet::getter(fn completed)]
+	pub(super) type Completed<T: Config> = StorageValue<_, bool, ValueQuery, OnCompletedEmpty>;
```

Comment:
So, now we are calling this storage entry after a boolean status (completed or not). There's nothing wrong with that, but as I came across in the code below, [the boolean blindness ](https://existentialtype.wordpress.com/2011/03/15/boolean-blindness/) becomes easily in the way.

My _suggestion_ here would be to rename this storage entry to `Status` (as in, _Migration status_), which would hold a value of type `Status`:

``` rust
 enum Status {
    Inactive,
    Ongoing, // If we need to cover this intermediate state
    Complete,
}
```

Then the `#[pallet::getter(fn completed)]` would become `#[pallet::getter(fn status)]`, and in the `ensure` expressions below we would have 

``` rust
ensure!(
  Self::status() != Status.Completed,
	Error::<T>::MigrationAlreadyCompleted
);
```

Reply:
I like. Will change that. I also don't like booleans, when I am not too lazy.